src/main/js/components/columns/ColumnEntry.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import AmpersandMixin from 'ampersand-react-mixin'
+import ActionCreator from '../../actions/ActionCreator'
+import OutsideClick from 'react-onclickoutside'
+import ColumnEditorOverlay from '../overlay/ColumnEditorOverlay'
+import i18n from 'i18next'
+import * as AccessControl from '../../helpers/accessManagementHelper'
+
+const ColumnEntry = React.createClass({
+  mixins: [AmpersandMixin, OutsideClick],
+
+  PropTypes: {
+    description: React.PropTypes.string.isRequired,
+    columnContent: React.PropTypes.array.isRequired,
+    index: React.PropTypes.number.isRequired,
+    selected: React.PropTypes.number.isRequired,
+    clickHandler: React.PropTypes.func.isRequired,
+    cancelEdit: React.PropTypes.func.isRequired,
+    langtag: React.PropTypes.string.isRequired,
+    name: React.PropTypes.string.isRequired,
+    readOnly: React.PropTypes.bool
+  },
+
+  handleClickOutside(evt) {
+    this.props.blurHandler()
+  },
+
+  getInitialState() {
+    return {
+      name: this.props.name,
+      description: this.props.description
+    }
+  },
+
+  handleClick() {
+    //only admin may modify columns
+    if (!AccessControl.isUserAdmin()) return
+
+    const {index,selected} = this.props
+    const letParentHandleClick = this.props.clickHandler
+    if (index === selected) {
+      this.editColumn()
+    } else {
+      letParentHandleClick()
+    }
+  },
+
+  handleInput(inputState) {
+    const self = this
```

Comment:
Self is not necessary, but can't simplify the return because of the way it needs to be invoked. Open for suggestions, that's a problem.